include/zephyr/drivers/display.h

```diff
@@ -101,6 +103,34 @@ enum display_orientation {
 	DISPLAY_ORIENTATION_ROTATED_270, /**< Rotated 270 degrees clockwise */
 };
 
+/** @brief Structure holding common display configuration. */
+struct display_common_config {
+	struct gpio_dt_spec backlight_gpio;
```

Comment:
It would be nice to have compile-time guard to save resources like:

```c
#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(backlight_gpio)
	struct gpio_dt_spec backlight_gpio;
#endif
```

And same for `pwm`

Reply:
The problem is that this is common code so you can't really use it, you'd end up with different versions of the struct depending on what compatible this is computed with. If you want to keep this compact you can use pointer to an external struct and only instantiate the struct that is used, then instead of 16+8 bytes for the pwm+gpio struct you'd either use 2+16 or 2+8 for the pointer plus the relevant struct.